atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                if value is not None:
+                    self._system_info[sec_key][box_key] = [value] * n_frames
+
+            box_keys = {'lattice_vectors': 'system'}
+            for box_key, sec_key in box_keys.items():
+                value = self._data_parser.hdf5_getter(box, self._nomad_to_box_group_map[box_key])
+                if value is not None:
+                    self._system_info[sec_key][box_key] = value
+                else:
+                    continue
+
+                if type(self._system_info[sec_key][box_key]) == h5py.Group:
+                    self._system_info[sec_key][box_key] = self._data_parser.hdf5_getter(self._system_info[sec_key][box_key], 'value')
+                    if self._system_info[sec_key][box_key] is None:
+                        continue
+                    elif len(self._system_info[sec_key][box_key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of box vectors and positions is not supported. ' + box_key + ' values will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][box_key] = [self._system_info[sec_key][box_key]] * n_frames
+        return self._system_info
+
+    @property
+    def observable_info(self):
+        if self._observable_info is None:
+            observables_group = self.h5md_groups.get('observables')
+            if observables_group is None:
+                return
+
+            self._observable_info = {
+                'configurational': {},
+                'ensemble_average': {},
+                'correlation_function': {}
+            }
+
+            def get_observable_paths(observable_group, current_path, paths):
+                for obs_key in observable_group.keys():
+                    path = obs_key + '.'
+                    observable = self._data_parser.hdf5_getter(observable_group, obs_key)
+                    observable_type = self._data_parser.hdf5_getter(observable_group, obs_key).attrs.get('type')
+                    if not observable_type:
+                        paths = get_observable_paths(observable, current_path + path, paths)
+                    else:
+                        paths.append(current_path + path[:-1])
+
+                return paths
+
+            observable_paths = get_observable_paths(observables_group, current_path='', paths=[])
+
+            for path in observable_paths:
+                observable = self._data_parser.hdf5_getter(observables_group, path)
+                observable_type = self._data_parser.hdf5_getter(observables_group, path).attrs.get('type')
+                observable_name = path.split('.')[0]
+                observable_label = '-'.join(path.split('.')[1:]) if len(path.split('.')) > 1 else ''
```

Comment:
you do not need the if here simply have
```python
observable_label = '-'.join(path.split('.)[-1:])

Reply:
That doesn't quite work for what I am trying to achieve. when `path=group.name` it returns `name` which is what I want. but if `path=group`, I actually want to return an empty string, but yours returns `group`

Reply:
ah yes sorry.

Reply:
ok, you are right, not sure what happened the first time :upside_down_face: thanks! done.